packages/notionable/src/base-trade/sagas/vault-capacity.ts

```diff
@@ -0,0 +1,107 @@
+import { AccountDefinition, Registry, TokenBalance } from '@notional-finance/core-entities';
+import { filterEmpty } from '@notional-finance/util';
+import { Observable, combineLatest, distinctUntilChanged, map } from 'rxjs';
+import { VaultTradeState } from '../base-trade-store';
+import { selectedNetwork } from '../../global';
+
+
+export function vaultCapacity(
+    state$: Observable<VaultTradeState>,
+    account$: Observable<AccountDefinition | null>,
+    selectedNetwork$: ReturnType<typeof selectedNetwork>
+  ) {
+    return combineLatest([account$, state$, selectedNetwork$]).pipe(    
+        distinctUntilChanged(
+            ([, p], [, c]) =>
+              p.tradeType === c.tradeType &&
+              p.calculationSuccess === c.calculationSuccess &&
+              p.collateralBalance?.hashKey === c.collateralBalance?.hashKey &&
+              p.inputErrors === c.inputErrors &&
+              c.debtBalance === c.debtBalance &&
```

Comment:
This should compare on the hashKey instead of on the object itself, that will reduce the number of executions we do here.